### ISSUE TITLE
No diagnostics reported for struct cycle across instance and static fields

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#75832]
**Version Used**: 

**Steps to Reproduce**:

Compile the following:

```csharp
struct C<T>
{
    static D<T> x;
}

struct D<T>
{
    C<D<T>> y;
}
```

**Expected Behavior**:
Error reported for cycle. For instance, the pre-Roslyn compiler reports:
```
(3,17): error CS0523: Struct member 'C<T>.x' of type 'D<T>' causes a cycle in the struct layout
(8,13): error CS0523: Struct member 'D<T>.y' of type 'C<D<T>>' causes a cycle in the struct layout
```

**Actual Behavior**:
No errors reported.

`peverify.exe` fails with the resulting assembly:
```
>peverify c.dll

Microsoft (R) .NET Framework PE Verifier.  Version  4.0.30319.0
Copyright (c) Microsoft Corporation.  All rights reserved.

[token  0x02000004] Type load failed.
[token  0x02000005] Type load failed.
2 Error(s) Verifying c.dll
```

**Comment by AlekseyTs** [comment]
See also #75701

**Comment by cston** [comment]
Moving to Backlog since the compiler does not crash.

**Comment by gafter** [comment]
I don't see a cycle. `C<T>` is an empty struct type.

**Comment by gafter** [comment]
Related to this is the runtime bug https://github.com/dotnet/runtime/issues/5479, which the runtime decided not to fix and yet closed as fixed. I believer the PEVerify issue is the same bug.
